website/blog/2023-02-20-7.21.0.md

```diff
@@ -0,0 +1,135 @@
+
+TC39, the committee that standardizes JavaScript, [recently approved](https://github.com/babel/proposals/issues/86#issuecomment-1414202318) some changes to the [Decorators proposal](https://github.com/tc39/proposal-decorators/) based on feedback from the TypeScript team.
+- Decorators on exported classes can now come either before or after the `export` keyword, but not in both places at the same time:
+  ```js title="JavaScript"
+  // valid
+  @dec
+  export class A {}
+
+  // valid
+  export @dec class B {}
+
+  // invalid
+  @dec
+  export @dec class C {}
+  ```
+  This relaxed restriction aims at simplifying migration from the "_legacy_" decorators version, by reducing the syntactic differences.
+- The methods on the `context.access` object provided to decorators expect the target object as their first parameter, rather than as their `this` receiver:
+  ```js title="JavaScript"
+  let accessX;
+
+  function dec(desc, context) {
+    accessX = context.access;
+    return dec;
+  }
+
+  class A {
+    @dec #x = 1;
+  }
+
+  // old semantics
+  accessX.get.call(new A) === 1;
+
+  // new semantics
+  accessX.get(new A) === 1;
+  ```
+- `context.access` has a new `.has` method, to check if an object has the corresponding element. Continuing the example above:
+  ```js title="JavaScript"
+  // #x in new A
+  accessX.has(new A) === true;
+  ```
+
+You can enable this new version of the decorators proposal by setting the `version` option of `"@babel/plugin-proposal-decorators"` to `"2023-01"`:
+
+```json title="babel.config.json"
+{
+  "presets": ["@babel/preset-env"],
+  "plugins": [
+    ["@babel/plugin-proposal-decorators", { "version": "2023-01" }]
+  ]
+}
+```
+
+### TypeScript 5.0
+
+TypeScript 5.0, currently released as a beta prerelease, introduces two new syntactic features:
+- `const` modifiers for type parameters
+  ```ts title="TypeScript"
+  function getName<const T extends { name: string }>(user: T): T["name"] {
+      return user.name;
+  }
+
+  let name = getName({ name: "Babel" });
+  //  ^? inferred type: "Babel", instead of just string.
+  ```
+- `export type *` declarations
+  ```ts title="TypeScript"
+  export type * from "./mod";
+  export type * as ns from "./mod";
+  ```
+  Babel relies on this new syntax to safely remove the re-`export` declaration while compiling from TypeScript to JavaScript.
+
+Additionally, TypeScript 5.0 introduces support for the standard Decorators proposal, that you can enable in Babel using `@babel/plugin-proposal-decorators`.
+
+You can read more about the new TypeScript features in their [release post](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/)!
+
+### Experimental support for `.cts` configuration files
+
+If you installed `@babel/preset-typescript`, or if you are running Babel using `ts-node`, you can now use `babel.config.cts` as a Babel configuration file written in TypeScript and CommonJS. You can read more about this in the [documentation](TODO: link).
```

Comment:
Oops, there's one more TODO here.
Also we want to add a repl link for the decorator?

Reply:
Whops, I'll fix those soon :)